arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -249,6 +249,9 @@ private static Types.NestedField getPhysicalType(
   }
 
   private void allocateDictEncodedVector() {
```

Comment:
This file contains the complete fix for the leak. All other changes are testing related